source/server/server.cc

```diff
@@ -210,11 +210,12 @@ InstanceUtil::BootstrapVersion InstanceUtil::loadBootstrapConfig(
     ProtobufMessage::ValidationVisitor& validation_visitor, Api::Api& api) {
   const std::string& config_path = options.configPath();
   const std::string& config_yaml = options.configYaml();
+  const envoy::config::bootstrap::v2::Bootstrap& config_proto = options.configProto();
 
   // Exactly one of config_path and config_yaml should be specified.
-  if (config_path.empty() && config_yaml.empty()) {
-    const std::string message =
-        "At least one of --config-path and --config-yaml should be non-empty";
+  if (config_path.empty() && config_yaml.empty() && config_proto.ByteSize() == 0) {
+    const std::string message = "At least one of Options::configPath, Options::configYaml or "
```

Comment:
This error message is not going to mean a lot to most people. Can you retain roughly the old text but expand if necessary?

Reply:
Done.